moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1718,9 +1715,11 @@ bool PlanningScene::shapesAndPosesFromCollisionObjectMessage(const moveit_msgs::
 
   bool switch_object_pose_and_shape_pose = false;
   if (num_shapes == 1)
-    if (object_pose.isApprox(Eigen::Isometry3d::Identity()))
+    if (moveit::core::isEmpty(object.pose))
```

Comment:
I had just accepted that identity object poses will always be overwritten by shape poses, but it's true that you might want to e.g. shift your mesh without changing the object pose itself. Nice!